### PR TITLE
Cancel

### PR DESCRIPTION
**Pull request by acesnik**
Added cancel button to main window and functionality to cancel Calibration, Search, and GPTMD tasks. Cancel functionality was added to XL tasks, but the effectiveness was not tested. Neo was ignored.

### CODE REVIEW

File: GUI/MainWindow.xaml.cs [hunk 1]
```diff
@@ -1255,6 +1258,17 @@ private bool SpectraFileExists(ObservableCollection<RawDataForDataGrid> rDOC, Ra
             return false;
         }
 
+        private void CancelButton_Click(object sender, RoutedEventArgs e)
+        {
+            string grammar = StaticTasksObservableCollection.Count <= 1 ? "this task" : "these tasks";
```
Comment (ianhirsch) [inline GUI/MainWindow.xaml.cs]: nice addition

Comment (acesnik) [reply GUI/MainWindow.xaml.cs]: Thanks!

File: GUI/MainWindow.xaml.cs [hunk 2]
```diff
@@ -1255,6 +1258,17 @@ private bool SpectraFileExists(ObservableCollection<RawDataForDataGrid> rDOC, Ra
...
+            if (MessageBox.Show("Are you sure you want to cancel " + grammar + "?", "Cancel Tasks", MessageBoxButton.OKCancel) == MessageBoxResult.OK)
+            {
+                GlobalVariables.StopLoops = true;
+                CancelButton.IsEnabled = false;
+                notificationsTextBox.AppendText("Canceling...\n");
```
Comment (zrolfs) [inline GUI/MainWindow.xaml.cs]: I like this, but it didn't go away after it finished when I canceled during protein loading. Maybe another message saying "Canceled!" or just remove the "Canceling..." once it's finished?

**Comment by zrolfs** [comment]
If you're looking to spend even more time on this, you could change the message next to the task progress from "Done!" to "Canceled!" and make the bar red instead of green :)

**Comment by acesnik** [comment]
@zrolfs and @ianhirsch Thanks for testing the feature! Would you like to commit to this branch to add/fix those features?

**Comment by zrolfs** [comment]
Not at the moment; I'm working on my projects 🔥

**Pull request merged**